### ISSUE TITLE
Do not hardcode seofuncs - plugin

### ISSUE THREAD

**Issue by TN03** [cmsimple-xh/cmsimple-xh#448]
https://github.com/cmsimple-xh/cmsimple-xh/blob/5439456c1442af0b22aee7c2467159de954945ce/cmsimple/seofuncs.php#L31

Wäre es nicht besser, wenn dar Name des Plugins aus einer (eventuell versteckten) Konfigurations-Variable käme?

**Comment by olape-git** [comment]
Auf jeden Fall. Das ließe alternative Lösungen zu. Ein guter Gedanke.

**Comment by cmb69** [comment]
Ja, aber dann sollte auch klar festgelegt werden, wie das Plugin mit dem Core interagieren kann.

**Comment by olape-git** [comment]
Na im Prinzip sind das nur `$redir, $scheme, $host, $path, $query_str`

Wäre vielleicht auch ein Ansatz für CleanURLs, wenn es nicht unbedingt in den Core soll.
Aber mit der Weile bin ich mir gar nicht mehr sicher, ob der Aufwand für CleanURLs überhaupt noch zu rechtfertigen ist.

**Comment by cmb69** [comment]
Hm, ich denke, dass seofuncs.php und adc_core/seofuncs.php viel zu stark gekoppelt sind, weil eben aus einer Funktion (`XH_URI_Cleaning()`) heraus ein PHP-Skript inkludiert wird, was dann eben vollen Zugriff auf alle lokalen Variablen der Funktion hat, und alle auf Skriptebene definierten Variablen, dann auch in der Funktion verfügbar sind. Und dass der Austausch der relevanten Information über eine handvoll Variablen erfolgt, finde ich auch nicht gut; man kann das natürlich dokumentieren, aber da geht schnell was übersehen (ein Tippfehler reicht ja schon, und rutscht vielleicht beim Testen durch).

Eigentlich geht es ja nur darum, dass adc_core (oder ein alternatives Plugin) die vorbelegten Informationen bekommt, und die Möglichkeit hat, diese zu ändern. Dazu könnte man einfach eine Funktion aufrufen, die Informationen als Argumente übergeben, und dann den Rückgabewert der Funktion in seofuncs.php aufdröseln. Das entkoppelt, und ist per se einfacher zu dokumentieren.

Ich würde es sogar bevorzugen, wenn nicht eine Funktion, sondern eine Klasse genutzt wird, die ein vom Core deklariertes Interface implementieren muss; das vermindert Inkompatibilitäten, und ist wohl noch leichter zu dokumentieren. Außerdem kann dann der Autoloader genutzt werden, so dass nicht mehr manuell inkludiert werden muss.

Aber egal ob nun Klasse/Interface, Funktion, oder so wie es jetzt ist, sobald wir das Plugin konfigurierbar machen, sind wir daran gebunden. Im Nachhinein bin ich z.B. über die Editor-Integration nicht glücklich; das hätte gewiss besser gelöst werden können, aber ändern können wir es nicht mehr so einfach.
    
Daher würde ich vorschlagen, diesen Punkt doch lieber auf 1.8 zu verschieben, um einen Schnellschuss zu vermeiden.

**Comment by olape-git** [comment]
Es muss nicht in die 1.7.6.
In der 1.8 steht dann eh drin, die Funktionalität von adc in XH zu übernehmen.
Das hätte ich das letzte Mal schon gern getan, wurde aber überstimmt.

Deiner Erklärung kann ich nur zum Teil folgen. :-(
Es ging mir hauptsächlich darum, dass das Plugin so früh wie möglich eingreift (und man das Plugin trotzdem als normales Plugin handeln kann),
ohne das XH erst große Teile abarbeitet und dann doch ein Redirect auslöst.
Wenn wir irgendwie die Reihenfolge der Plugins beeinflussen könnten, ausser nur alphabetisch, dann hätte mir das auch gereicht.
